src/System.Linq/tests/SingleTests.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System.Linq.Tests.Helpers;
+using Xunit;
+
+namespace System.Linq.Tests
+{
+    public class SingleTests
+    {
+        [Fact]
+        public void FailOnEmpty()
+        {
+            Assert.Throws<InvalidOperationException>(() => Enumerable.Empty<int>().Single());
+        }
+
+        [Fact]
+        public void DefaultOnEmpty()
+        {
+            Assert.Equal(0, Enumerable.Empty<int>().SingleOrDefault());
+        }
+
+        [Fact(Skip="Runs for thousands of years. Then throws wrong exception")]
+        public void FailOnManyMatches()
+        {
+            Assert.Throws<InvalidOperationException>(() => Enumerable.Range(0, int.MaxValue).Single(i => i > 0));
```

Comment:
Since #2350 has a set of tests that work before the change and fail after, and a set of tests that fail before the change and work after, then whatever decision is made there's a set of tests that will enforce that decision.

The tests here couldn't be used as-is anyway; take off the `Skip` and all they would test is how well your CI server holds up against people trying to use tests to DoS it!